### PR TITLE
Added default dotnet-tools + additional tasks to launch them

### PR DESCRIPTION
**Pull request by vzarytovskii**
Added tools I use the most. Now I need to manually install them in dev container every time, this will allow installing them in one command.

Also, added a task to launch fsi/fsc under trace, so it can be opened with speedscope later.

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -2,6 +2,25 @@
...
+        {
+            "id": "fsiArgsPrompt",
+            "description": "Enter arguments for fsi",
+            "default": "",
+            "type": "promptString",
+        },
+        {
+            "id": "PickProcess",
+            "description": "Enter process id",
+            "type": "promptString"
```
[the remark this answers is not in the document]
Comment (vzarytovskii) [reply .vscode/tasks.json:21]: Ah, good point

File: .config/dotnet-tools.json [hunk 2]
```diff
@@ -1,12 +1,36 @@
 {
-  "version": 1,
   "isRoot": true,
   "tools": {
+    "dotnet-counters": {
```
[the remark this answers is not in the document]
Comment (vzarytovskii) [reply .config/dotnet-tools.json:4]: Wait, what's the issue? They're not restored globally, so should be fine for dev containers?

[the remark this answers is not in the document]
Comment (baronfel) [reply .config/dotnet-tools.json:4]: No problem - just a comment that the SDK tools don't have a nice command to update all versions in one command. A UX problem for the SDK team, no problem here on your end :)

[the remark this answers is not in the document]
Comment (vzarytovskii) [reply .config/dotnet-tools.json:4]: Oh, I see.

File: .vscode/launch.json [hunk 3]
```diff
@@ -17,17 +17,17 @@
     ],
     "configurations": [
         {
-            "name": "Launch FSI (Debug, .NET 7.0)",
+            "name": "Launch FSI",
             "type": "coreclr",
             "request": "launch",
             // TODO: Shall we assume that it's already been built, or build it every time we debug?
-            "preLaunchTask": "Build (Debug)",
+            "preLaunchTask": "${defaultBuildTask}",
```
Comment (psfinaki) [inline .vscode/launch.json:24]: Where's this defined?

**Review by psfinaki — COMMENTED** [review]
This looks useful - how about we mention some of those tasks in the DEVGUIDE?

**Comment by vzarytovskii** [comment]
> This looks useful - how about we mention some of those tasks in the DEVGUIDE?

Probably not worth it. These are standard tools, you either use them or don't. We don't mention VS diagnostic tools or dotTrace/dotMemory/windbg there.

File: .vscode/launch.json (hunk 3 above)
Comment (vzarytovskii) [reply .vscode/launch.json:24]: In vscode

File: .vscode/tasks.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (vzarytovskii) [reply .vscode/tasks.json:21]: Type can only be "process" or "shell", and neither have pid picker. Afaik, only `launch.json` can pick a process.

**Comment by psfinaki** [comment]
> Probably not worth it? These are standard tools/tasks, you either use them or don't. I feel like all of them are self explanatory?
> We don't mention how to launch with VS diagnostic tools or dotTrace/dotMemory/windbg there.
 
Mmm e.g. we have that note about updating XLFs and it says "This can be done automatically by running this command" and I just thought we could add smth like "... or by executing this task in VSCode". So that people have a notion of those.

I personally don't use VSCode tasks so I don't know - but now looking at the PR I am thinking it is easy enough to give them a try and change my dev routines.

Up to you ofc.

**Comment by vzarytovskii** [comment]
> > Probably not worth it? These are standard tools/tasks, you either use them or don't. I feel like all of them are self explanatory?
> > We don't mention how to launch with VS diagnostic tools or dotTrace/dotMemory/windbg there.
> 
> Mmm e.g. we have that note about updating XLFs and it says "This can be done automatically by running this command" and I just thought we could add smth like "... or by executing this task in VSCode". So that people have a notion of those.
> 
> I personally don't use VSCode tasks so I don't know - but now looking at the PR I am thinking it is easy enough to give them a try and change my dev routines.
> 
> Up to you ofc.

I wouldn't update them just for now. Since our VSCode support is not great at the moment. I'd update it and fully add once we do more LSP testing, I think.

**Review by psfinaki — APPROVED** [review]

**Review by abonie — APPROVED** [review]

**Pull request merged**